### ISSUE TITLE
Crash when switching board

### ISSUE THREAD

**Issue by josaur** [stefan-niedermann/nextcloud-deck#930]
With the newest Version an Issue occurred. Every time I'm changing the Board the App crashes. The previous Versions worked fine. 

```
Full Crash:

App Version: 1.16.2
App Version Code: 1016002
App Flavor: fdroid

Files App Version Code: 30140190

---

OS Version: 3.4.113-ge2b4a42(daeb3d5f6c)
OS API Level: 28
Device: s3ve3g
Manufacturer: samsung
Model (and Product): GT-I9301I (lineage_s3ve3gxx)

---

java.lang.ExceptionInInitializerError
	at f.a.a.c.d$a.a(Unknown Source:0)
	at f.a.a.c.d.c(Unknown Source:27)
	at androidx.viewpager2.widget.b.c(Unknown Source:18)
	at androidx.viewpager2.widget.b.c(Unknown Source:18)
	at androidx.viewpager2.widget.e.d(Unknown Source:4)
	at androidx.viewpager2.widget.e.m(Unknown Source:23)
	at androidx.viewpager2.widget.ViewPager2.k(Unknown Source:84)
	at androidx.viewpager2.widget.ViewPager2.j(Unknown Source:6)
	at it.niedermann.nextcloud.deck.ui.MainActivity.J0(Unknown Source:8)
	at it.niedermann.nextcloud.deck.ui.MainActivity.K0(Unknown Source:0)
	at it.niedermann.nextcloud.deck.ui.f1.run(Unknown Source:8)
	at android.app.Activity.runOnUiThread(Activity.java:6283)
	at it.niedermann.nextcloud.deck.ui.MainActivity.z1(Unknown Source:140)
	at it.niedermann.nextcloud.deck.ui.MainActivity.A1(Unknown Source:0)
	at it.niedermann.nextcloud.deck.ui.e1.a(Unknown Source:4)
	at androidx.lifecycle.LiveData.d(Unknown Source:29)
	at androidx.lifecycle.LiveData.e(Unknown Source:44)
	at androidx.lifecycle.LiveData.p(Unknown Source:14)
	at androidx.lifecycle.x.p(Unknown Source:0)
	at androidx.lifecycle.f0$c.a(Unknown Source:27)
	at androidx.lifecycle.v$a.a(Unknown Source:20)
	at androidx.lifecycle.LiveData.d(Unknown Source:29)
	at androidx.lifecycle.LiveData.e(Unknown Source:44)
	at androidx.lifecycle.LiveData.p(Unknown Source:14)
	at androidx.lifecycle.LiveData$a.run(Unknown Source:18)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.IllegalStateException: java.lang.NoSuchMethodException: selectTab [class com.google.android.material.tabs.TabLayout$g, boolean]
	at f.a.a.c.d$a.b(Unknown Source:12)
	at f.a.a.c.d$a.<clinit>(Unknown Source:17)
	... 32 more
Caused by: java.lang.NoSuchMethodException: selectTab [class com.google.android.material.tabs.TabLayout$g, boolean]
	at java.lang.Class.getMethod(Class.java:2068)
	at java.lang.Class.getDeclaredMethod(Class.java:2047)
	at f.a.a.c.d$a.b(Unknown Source:0)
	... 33 more

**Comment by stefan-niedermann** [comment]
Is already fixed and [released in `1.16.3`](https://github.com/stefan-niedermann/nextcloud-deck/releases/tag/1.16.3).

I can't do anything more than wait for F-Droid unfortunately. Use an older version as workaroundy, sorry.

It was caused by an [attempt to reduce the size of the app](https://github.com/stefan-niedermann/nextcloud-deck/commit/915d413a196d262ac7a069e886086814add4f888).

**Comment by josaur** [comment]
Sorry for the duplicate and thanks for the reply and the fix

**Comment by stefan-niedermann** [comment]
Wasn't exactly a duplicate - the error was in the Notes **and** Deck app :slightly_smiling_face:

**Comment by varac** [comment]
> I can't do anything more than wait for F-Droid unfortunately. Use an older version as workaroundy, sorry.

Is there a way to watch the status of the fdroid build ? Downgrade to 1.6.1 fails for me unfortnately, and I don't want to remove/install an older version because I fear I might loose mobile cards that aren't correctly synced with 1.6.2.

**Comment by stefan-niedermann** [comment]
Sounds reasonable. As far as i can see, [the build has been finished](https://f-droid.org/wiki/index.php?title=it.niedermann.nextcloud.deck/lastbuild&redirect=no) by F-Droid, but i don't know how to watch an index update.
The last one was two days ago, so i personally expect the next index update to be tomorrow or in two days.

Please note that `1.17.2` and `1.17.3` already are known to *possibly* cause trouble for *some* users, depending on their device, manfucturer and software version. If you want to go absolutely sure, wait for `1.17.4` (also tagged but not yet built by F-Droid).

Sorry all for the inconveniences. At the Play Store i have more control about beta channels, staged rollout and cancelling releases in case we notice an error. But because of the decentralized nature and little resources for the build infrastructure this kind of control is not available for F-Droid.

**Comment by varac** [comment]
I just upgraded to 1.17.2 and got this (probably already fixed by newer releases that are still not available i fdroid ?):

```
Full Crash:

App Version: 1.17.2
App Version Code: 1017002
App Flavor: fdroid

Files App Version Code: 30150190

---

OS Version: 4.4.205-g240e125(2b63a9093e)
OS API Level: 30
Device: H3113
Manufacturer: Sony
Model (and Product): H3113 (H3113)

---

java.lang.RuntimeException: Exception while computing database live data.
	at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:92)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
Caused by: android.database.sqlite.SQLiteException: table "Board" already exists (code 1 SQLITE_ERROR): , while compiling: CREATE TABLE "Board" (`localId` INTEGER PRIMARY KEY AUTOINCREMENT, `accountId` INTEGER NOT NULL, `id` INTEGER, `status` INTEGER NOT NULL, `lastModified` INTEGER, `lastModifiedLocal` INTEGER, `title` TEXT, `ownerId` INTEGER NOT NULL, `color` INTEGER, `archived` INTEGER NOT NULL, `shared` INTEGER NOT NULL, `deletedAt` INTEGER, `permissionRead` INTEGER NOT NULL, `permissionEdit` INTEGER NOT NULL, `permissionManage` INTEGER NOT NULL, `permissionShare` INTEGER NOT NULL, `etag` TEXT, FOREIGN KEY(`ownerId`) REFERENCES `User`(`localId`) ON UPDATE NO ACTION ON DELETE CASCADE , FOREIGN KEY(`accountId`) REFERENCES `Account`(`id`) ON DELETE CASCADE )
	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1045)
	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:652)
	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
	at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:33)
	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1919)
	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1841)
	at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.execSQL(FrameworkSQLiteDatabase.java:242)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.migration.Migration_30_31.migrate(Migration_30_31.java:43)
	at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:99)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:177)
	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:416)
	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:316)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:145)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:106)
	at androidx.room.RoomDatabase.inTransaction(RoomDatabase.java:476)
	at androidx.room.RoomDatabase.assertNotSuspendingTransaction(RoomDatabase.java:281)
	at androidx.room.RoomDatabase.query(RoomDatabase.java:324)
	at androidx.room.util.DBUtil.query(DBUtil.java:83)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.dao.AccountDao_Impl$5.call(AccountDao_Impl.java:276)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.dao.AccountDao_Impl$5.call(AccountDao_Impl.java:273)
	at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:90)
	... 3 more

```

**Comment by stefan-niedermann** [comment]
@varac we are aware of this kind of errors and currently try to fix it in #935 (your specific error case should already been fixed in `1.17.4` or so, but there might possibly follow more).

You can either wait for further `1.17.x` releases or [clear the storage of the app](https://github.com/stefan-niedermann/nextcloud-deck/blob/master/FAQ.md#i-have-experienced-an-error) and readd your account (unsynchronized changes will be lost this way).
